app/security_system.py

```diff
@@ -66,13 +67,36 @@ def recognize_face(self, frame):
 
         # TODO 3
         # Perform face detection
+        faces = face_cascade.detectMultiScale(gray_frame, scaleFactor = 1.2, minNeighbors = 5, minSize = (30,30))
         # Iterate through each detected face
-        # Extract the detected face region
-        # Perform facial recognition on the detected face
-        # Draw bounding box around the detected face
-        # Annotate the frame with the recognized user's name and authorization status
-        # Log access attempt
-        # Your code goes here
+        for face in faces:
+            # Extract the detected face region
+            x,y,w,h = face
+            X,Y = x+w,y+h
+            image = gray_frame[x:X, y:Y]
+            # Perform facial recognition on the detected face
+            label, confidence = self.face_recognizer.predict(image)
+            # Draw bounding box around the detected face
+            l= 20
+            t= 5
+            cv2.rectangle(frame,face,(255,0,200),1)
```

Comment:
Can we draw bounding boxes with different colors for different authorization statuses? For example: Green if authorized and red if not.